src/Hl7.FhirPath/FhirPath/IValueProviderFPExtensions.cs

```diff
@@ -50,61 +45,59 @@ private static CompiledExpression getCompiledExpression(string expression)
 
                 _mruList.Remove(expression);
                 _mruList.Add(expression);
-                
+
                 return ce;
             }
         }
 
 
-        public static IEnumerable<IElementNavigator> Select(this IElementNavigator input, string expression, EvaluationContext ctx=null)
+        public static IEnumerable<ITypedElement> Select(this ITypedElement input, string expression, EvaluationContext ctx = null)
         {
             var evaluator = getCompiledExpression(expression);
             return evaluator(input, ctx ?? EvaluationContext.CreateDefault());
         }
 
-        [Obsolete("Replace with the overload taking an EvaluationContext, initialized with the resource parameter")]
-        public static IEnumerable<IElementNavigator> Select(this IElementNavigator input, string expression, IElementNavigator resource)
+        [Obsolete("Use Select(this ITypedElement input) instead")]
+        public static IEnumerable<IElementNavigator> Select(this IElementNavigator input, string expression, EvaluationContext ctx = null)
         {
-            return input.Select(expression, new EvaluationContext(resource));
+            return Select(input.ToTypedElement(), expression, ctx).Select(t => t.ToElementNavigator());
```

Comment:
This is a bit odd, in converting the output back to an `IElementNavigator` the node type is lost. So I don't think it is possible to cast to the node type to access things like `ShortPath`
see: https://github.com/ewoutkramer/fhir-net-api/pull/714/files#diff-9a88142c6637196a80e58ef1714b96fbL115